draft-ietf-kitten-krb-spake-preauth-04.xml

```diff
@@ -565,25 +545,32 @@ KEY_USAGE_SPAKE_TRANSCRIPT              66
       slightly from the method used to generate K' in Section 3 of <xref
       target="I-D.irtf-cfrg-spake2">SPAKE</xref>.</t>
 
-      <t>An input string is assembled by concatenating the following values:
+      <t>Each derived key K'[n] has the same encryption type as the initial
+      reply key. To compute its contents, the hash function associated with
+      the selected group is computed over the concatenation of the following
+      values:
       <list style="symbols">
         <t>The fixed string "SPAKEkey".</t>
-        <t>The group number as a big-endian four-byte unsigned binary
+        <t>The group number as a big-endian four-byte two's complement binary
         number.</t>
-        <t>The encryption type of the initial reply key as a
-        big-endian four-byte unsigned binary number.</t>
+        <t>The encryption type of the initial reply key as a big-endian
+        four-byte two's complement binary number.</t>
+        <t>The PRF+ output used to compute the initial secret input w as
+        specified in <xref target="spakeparams"/>.</t>
         <t>The SPAKE result K, converted to an octet string as specified in
         <xref target="spakeparams"/>.</t>
-        <t>The transcript checksum.</t>
+        <t>The transcript hash.</t>
         <t>The KDC-REQ-BODY encoding for the request being sent or responded
         to. Within a FAST channel, the inner KDC-REQ-BODY encoding MUST be
         used.</t>
         <t>The value n as a big-endian four-byte unsigned binary number.</t>
+        <t>A single-byte block counter, with the initial value 0x01.</t>
       </list>
-      The derived key K'[n] has the same encryption type as the initial reply
-      key, and has the value random-to-key(PRF+(initial-reply-key,
-      input-string)). PRF+ is defined in <xref target="RFC6113"/> section
-      5.1.</t>
+      If the hash output is too small for the encryption type's key generation
+      seed length, the block counter value is incremented and the hash
+      function re-computed to produce as many blocks as are required. The
+      result is truncated to the key generation seed length, and the
+      random-to-key function is used to produce the key value.</t>
```

Comment:
Do we have test vectors that exercise this?

Reply:
Only insofar as they test that the initial count is 1.  Having test vectors for multiple blocks would be difficult.  By having it be a single byte and verifying the initial counter, I hope to avoid the pitfalls we saw with GSS PRF.

Reply:
And we don't want to try and do a dummy test vector with SHA1+AES256 or even just some raw intermediate state that has more output, to ensure it gets exercised?

Reply:
I added an AES256-edwards25519-SHA1 test vector using my Python code.  Exercising it in the production C implementation would be tough, as it would require parameterizing the key derivation function in an unnatural way.  I expect other implementors would run into similar difficulties making use of this test vector, but we can still include it.

Reply:
Thanks, it makes me feel better to know that this is not totally untested.